### ISSUE TITLE
[Feature Request] Mob leveling system

### ISSUE THREAD

**Issue by Balakay420** [CleverNucleus/PlayerEx#78]
Every time I play MC with friends, we run into the same issue. It gets too easy. After a certain point, mobs do not pose a threat. Especially after stat increases and weapon/armor enhancements. 

I'm not sure if this is possible to integrate into PlayerEx or if it would have to be an add-on, but my idea is this.

A leveling system for mobs. The higher it's level is, the more increased heath, damage,armor, and loot it has. Along with any other attributes that could be used to make them more "difficult".

Levels could either be randomly assigned at spawn OR it could be relative to the players level. Average level if multiplayer.

**Comment by CleverNucleus** [comment]
Hi there,
I completely understand where you're coming from, and I feel the same way. I'm aware of [this](https://github.com/Flemmli97/ImprovedMobs/issues/165) particular suggestion, and hope that something will come of it. It is likely, however, that I will at some point begin working on a levelling system mod for mobs - details pending, albeit area-averaged player level as a predicate for mob difficulty would be ideal.

For now though, I am not actively working on something like this.
Thanks.

**Comment by Balakay420** [comment]
My bad I didn't realize I already put an issue in for this.
I may have found a mod that could implement this, I would greatly appreciate it if you could take a look at it, and tell me if it's possible or not
https://www.curseforge.com/minecraft/mc-mods/power-scale

**Comment by Candy-Poppi** [comment]
I personally use 3 (or more) mods to boost mobs.

[Farsighted](https://www.curseforge.com/minecraft/mc-mods/farsighted-mobs) : Mobs see you from further away.
[Eldritch](https://www.curseforge.com/minecraft/mc-mods/eldritch-mobs) : Sometimes mob can have some properties, like "mini boss".
[RPGDifficulty](https://www.curseforge.com/minecraft/mc-mods/rpgdifficulty) : More further you go from spawn, more harder they are. You can also use time!

You can take look to "Dusk" mod, and "Bloodmoon" ;)

**Comment by bugescape** [comment]
Its not just mob, its about the fact that you dont cap your leveling system makes you literally able to grind towards superman, there is no balance consideration at all.

Due to the fact that this is the single mod that so popular and also it is the one that can cause exploit to end game fun and server IMBA, suggest to implement below simple solutions:

1, significantly increase the expanse of xp to level up (i am talking about 10000x more expensive at least, you cant believe how sweaty some players are), and 

2, cap the level to a maximum at 30, 

3, staying at each level consume you more and more very expensive energy/xp/special food or sort of thing all the time, just like you need protein to maintain muscle, and you will need to consume gold to maintain supermanness...this forces you to build more and more farm automatic farm to maintain your skill, otherwise you get slowness or weakness or sort of thing, or even level down

4, the luck attribute is also too powerful, please just make it act like fortune III and plus and so on. Luck cause rare item to drop end up forcing people to just thrown those items into lava

**Comment by CleverNucleus** [comment]
Hi @bugescape ,

I appreciate your input; however, your simple solutions 1, 2 and 4 are already implemented and/or have a solution: 

1. The level formula can be completely customised in the config.
2. The maximum level can be easily capped to any value using a datapack.
4. The luck attribute is a _vanilla_ attribute, and the amount you accrue over the course of your game by levelling can be completely configured, or disabled entirely such that you do not gain anything in the luck attribute.

Regarding your third suggestion, and indeed your overall point, I reiterate what has been said previously - that a mob levelling system is needed, and it is only the lack of mobs being able to get stronger too that makes this mod unbalanced at times. There is also an argument to be made about other facets of the game that may currently become stale; namely, environmental damage such as lava, fire, or fall damage - but I think that is out of scope for this issue.

Tl;dr; it really is just about a mob levelling system.

**Comment by Hokmoon** [comment]
> Its not just mob, its about the fact that you dont cap your leveling system makes you literally able to grind towards superman, there is no balance consideration at all.
> 
> Due to the fact that this is the single mod that so popular and also it is the one that can cause exploit to end game fun and server IMBA, suggest to implement below simple solutions:
> 
> 1, significantly increase the expanse of xp to level up (i am talking about 10000x more expensive at least, you cant believe how sweaty some players are), and
> 
> 2, cap the level to a maximum at 30,
> 
> 3, staying at each level consume you more and more very expensive energy/xp/special food or sort of thing all the time, just like you need protein to maintain muscle, and you will need to consume gold to maintain supermanness...this forces you to build more and more farm automatic farm to maintain your skill, otherwise you get slowness or weakness or sort of thing, or even level down
> 
> 4, the luck attribute is also too powerful, please just make it act like fortune III and plus and so on. Luck cause rare item to drop end up forcing people to just thrown those items into lava

I am just curious cause my friend is using this mod on a new server. Does the Melee crit chance from luck make ANY attack have a percentage chance to crit? Or Does it give a higher chance to crit when you are doing a jump attack?

**Comment by CleverNucleus** [comment]
Hi @Hokmoon, 
it is any melee attack, regardless of jumping. That also means that without any melee crit chance, you cannot crit even when jumping.

**Comment by Hokmoon** [comment]
> Hi @Hokmoon, it is any melee attack, regardless of jumping. That also means that without any melee crit chance, you cannot crit even when jumping.

Thank you!

**Comment by liabri** [comment]
I don't think anything can be this about this on PlayerExs end, at least realistically. Mods like RpgDifficulty would have to be the ones to take ones PlayerEx level into account when spawning mobs. Is there an issue on their repo regarding playerex support ?

**Comment by bugescape** [comment]
> 

can you show how do you cap it using datapack?

**Comment by bugescape** [comment]
can i cap it using datapack on server side only? because i am running a large scale server

**Comment by liabri** [comment]
Not sure how your proposition would help @Elekjid, you're still going to end up too powerful relative to the mobs around. I think you're thinking of the getting too OP from just XP issue (which is exacerbated with mob farms, my suggestion for this is to base level on achievements rather then pure XP).

**Comment by bugescape** [comment]
just simply cap the level to level 50 why it is so hard to do? why cant you just provide the datapack? @Hokmoon

**Comment by emibanana** [comment]
@CleverNucleus

**Comment by emibanana** [comment]
please provide method to cap the leveling system

**Comment by CleverNucleus** [comment]
@emibanana You will have to be more specific. Do you want to set a maximum level that players cannot surpass? Or are you referring to the chunk-based experience system - in which case, cap how?

**Comment by emibanana** [comment]
> 

hi, would like to cap the player level to a allowed maximum value, it just says in your description that this can be done via datapack but we dont know how to make one to make affective on your mod

**Comment by CleverNucleus** [comment]
@emibanana Please see the [wiki](https://clevernucleus.github.io/playerex-1.18.2_faq.html#2-how-do-i-set-the-players-max-level).
